File: jobkorea_css/collected/jobkorea_css_collector.js
const { chromium } = require('playwright');
const fs = require('fs').promises;
const path = require('path');

class JobKoreaCSSCollector {
    constructor() {
        this.baseUrl = 'https://www.jobkorea.co.kr';
        this.cssUrls = new Set();
        this.inlineStyles = [];
        this.componentStyles = {};
        this.colorPalette = new Set();
        this.typography = {};
    }

    async extractCssUrls(page) {
        // Get <link> stylesheet URLs
        const linkElements = await page.$$('link[rel="stylesheet"]');
        for (const link of linkElements) {
            const href = await link.getAttribute('href');
            if (href) {
                const fullUrl = new URL(href, this.baseUrl).href;
                this.cssUrls.add(fullUrl);
            }
        }

        // Get @import URLs from style tags
        const styleElements = await page.$$('style');
        for (const style of styleElements) {
            const content = await style.innerText();
            const importMatches = content.match(/@import\s+url\(['"]\?([^'"]+)['"]\)/g) || [];
            for (const match of importMatches) {
                const url = match.match(/url\(['"]\?([^'"]+)['"]\)/)[1];
                const fullUrl = new URL(url, this.baseUrl).href;
                this.cssUrls.add(fullUrl);
            }
        }
    }

    async extractInlineStyles(page) {
        const styleElements = await page.$$('style');
        for (const style of styleElements) {
            const content = await style.innerText();
            if (content.trim()) {
                this.inlineStyles.push({
                    url: page.url(),
                    content: content
                });
            }
        }
    }

    extractColorsFromStyles(styles) {
        const colorProperties = ['color', 'backgroundColor', 'borderColor', 'borderTopColor',
                               'borderRightColor', 'borderBottomColor', 'borderLeftColor'];
        
        for (const prop of colorProperties) {
            if (styles[prop] && styles[prop] !== 'rgba(0, 0, 0, 0)' && styles[prop] !== 'transparent') {
                this.colorPalette.add(styles[prop]);
            }
        }
    }

    async extractComponentStyles(page, componentName, selector) {
        try {
            const elements = await page.$$(selector);
            if (!elements.length) return;

            const componentStyles = [];
            
            // Get styles from first few instances
            for (let i = 0; i < Math.min(3, elements.length); i++) {
                const element = elements[i];
                const styles = await element.evaluate((el) => {
                    const computed = window.getComputedStyle(el);
                    const styles = {};
                    
                    // Important style properties
                    const properties = [
                        'color', 'background-color', 'background-image',
                        'font-family', 'font-size', 'font-weight', 'line-height',
                        'padding', 'margin', 'border', 'border-radius',
                        'width', 'height', 'display', 'position',
                        'box-shadow', 'text-align', 'text-decoration'
                    ];
                    
                    properties.forEach(prop => {
                        const camelCase = prop.replace(/-([a-z])/g, (g) => g[1].toUpperCase());
                        styles[camelCase] = computed.getPropertyValue(prop);
                    });
                    
                    return styles;
                });
                
                this.extractColorsFromStyles(styles);
                
                // Extract typography info
                if (styles.fontFamily) {
                    const fontKey = `${styles.fontFamily}_${styles.fontSize}_${styles.fontWeight}`;
                    this.typography[fontKey] = {
                        fontFamily: styles.fontFamily,
                        fontSize: styles.fontSize,
                        fontWeight: styles.fontWeight,
                        lineHeight: styles.lineHeight
                    };
                }
                
                componentStyles.push(styles);
            }
            
            this.componentStyles[componentName] = componentStyles;
            
        } catch (error) {
            console.error(`Error extracting styles for ${componentName}:`, error.message);
        }
    }

    async collectPageStyles(page, pageName) {
        console.log(`\nCollecting styles from ${pageName}...`);
        
        // Extract CSS URLs and inline styles
        await this.extractCssUrls(page);
        await this.extractInlineStyles(page);
        
        // Define components to analyze based on page
        let components = {};
        
        if (pageName === 'main') {
            components = {
                'header': 'header, .header, #header',
                'navigation': 'nav, .nav, .navigation, .gnb',
                'search_box': '.search, .search-box, .search-form, input[type="search"]',
                'main_banner': '.banner, .main-banner, .visual',
                'job_card': '.job-card, .recruit-item, .list-item, .job-item',
                'button_primary': '.btn, .button, .btn-primary, button',
                'footer': 'footer, .footer, #footer'
            };
        } else if (pageName === 'job_list') {
            components = {
                'filter_sidebar': '.filter, .sidebar, .search-filter',
                'job_listing': '.list-item, .job-item, .recruit-info',
                'pagination': '.pagination, .paging, .page-nav',
                'sort_options': '.sort, .order, .sorting'
            };
        } else if (pageName === 'company') {
            components = {
                'company_card': '.company-item, .corp-item, .company-info',
                'company_logo': '.logo, .company-logo, .corp-logo',
                'info_section': '.info, .company-detail, .corp-detail'
            };
        } else if (pageName === 'job_detail') {
            components = {
                'job_header': '.detail-header, .job-header, .recruit-header',
                'job_content': '.content, .detail-content, .job-detail',
                'apply_button': '.apply, .btn-apply, .apply-btn',
                'company_info': '.company-info, .corp-info'
            };
        } else {
            components = {
                'general_card': '.card, .box, .panel',
                'form_input': 'input, textarea, select',
                'table': 'table, .table'
            };
        }
        
        // Extract component styles
        for (const [componentName, selector] of Object.entries(components)) {
            await this.extractComponentStyles(page, `${pageName}_${componentName}`, selector);
        }
    }

    async run() {
        const browser = await chromium.launch({ headless: true });
        const context = await browser.newContext({
            viewport: { width: 1920, height: 1080 },
            userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36'
        });
        const page = await context.newPage();
        
        // Pages to visit
        const pagesToVisit = [
            ['main', '/'],
            ['job_list', '/recruit/joblist'],
            ['company', '/company/companylist'],
            ['talent', '/talent/talentlist']
        ];
        
        for (const [pageName, pagePath] of pagesToVisit) {
            try {
                const url = new URL(pagePath, this.baseUrl).href;
                console.log(`\nVisiting ${pageName}: ${url}`);
                await page.goto(url, { waitUntil: 'networkidle', timeout: 30000 });
                await page.waitForTimeout(2000); // Wait for dynamic content
                
                await this.collectPageStyles(page, pageName);
                
                // If on job list page, try to get a job detail URL
                if (pageName === 'job_list') {
                    try {
                        const jobLink = await page.$('a[href*="/recruit/jobinfo"]');
                        if (jobLink) {
                            const jobUrl = await jobLink.getAttribute('href');
                            if (jobUrl) {
                                const fullJobUrl = new URL(jobUrl, this.baseUrl).href;
                                console.log(`\nVisiting job detail: ${fullJobUrl}`);
                                await page.goto(fullJobUrl, { waitUntil: 'networkidle', timeout: 30000 });
                                await page.waitForTimeout(2000);
                                await this.collectPageStyles(page, 'job_detail');
                            }
                        }
                    } catch (error) {
                        console.error('Could not visit job detail page:', error.message);
                    }
                }
                
            } catch (error) {
                console.error(`Error visiting ${pageName}:`, error.message);
            }
        }
        
        await browser.close();
        
        // Save results
        await this.saveResults();
    }

    async saveResults() {
        const outputDir = path.join('D:', 'wspace', 'jobkorea_css', 'collected');
        
        // Save CSS URLs
        const cssUrlsContent = '# CSS File URLs from JobKorea\n\n' + 
                             Array.from(this.cssUrls).sort().join('\n');
        await fs.writeFile(path.join(outputDir, 'css_urls.txt'), cssUrlsContent, 'utf-8');
        
        // Save inline styles
        await fs.writeFile(
            path.join(outputDir, 'inline_styles.json'),
            JSON.stringify(this.inlineStyles, null, 2),
            'utf-8'
        );
        
        // Save component styles
        await fs.writeFile(
            path.join(outputDir, 'component_styles.json'),
            JSON.stringify(this.componentStyles, null, 2),
            'utf-8'
        );
        
        // Save color palette
        const colorList = Array.from(this.colorPalette);
        const colorData = {
            all_colors: colorList.sort(),
            rgb_colors: colorList.filter(c => c.startsWith('rgb')),
            hex_colors: colorList.filter(c => c.startsWith('#')),
            named_colors: colorList.filter(c => !c.startsWith('#') && !c.startsWith('rgb'))
        };
        
        await fs.writeFile(
            path.join(outputDir, 'color_palette.json'),
            JSON.stringify(colorData, null, 2),
            'utf-8'
        );
        
        // Save typography
        await fs.writeFile(
            path.join(outputDir, 'typography.json'),
            JSON.stringify(this.typography, null, 2),
            'utf-8'
        );
        
        // Create summary report
        const summary = `JobKorea CSS Collection Summary
==============================

Total CSS URLs found: ${this.cssUrls.size}
Inline style blocks: ${this.inlineStyles.length}
Components analyzed: ${Object.keys(this.componentStyles).length}
Unique colors found: ${this.colorPalette.size}
Typography variations: ${Object.keys(this.typography).length}

Components analyzed:
${Object.keys(this.componentStyles).sort().map(c => `  - ${c}`).join('\n')}
`;
        
        await fs.writeFile(path.join(outputDir, 'summary.txt'), summary, 'utf-8');
        
        console.log('\n\nCollection complete! Results saved to:');
        console.log(`  - ${outputDir}/css_urls.txt`);
        console.log(`  - ${outputDir}/inline_styles.json`);
        console.log(`  - ${outputDir}/component_styles.json`);
        console.log(`  - ${outputDir}/color_palette.json`);
        console.log(`  - ${outputDir}/typography.json`);
        console.log(`  - ${outputDir}/summary.txt`);
    }
}

// Run the collector
const collector = new JobKoreaCSSCollector();
collector.run().catch(console.error);

File: jobkorea_css/collected/jobkorea_css_collector.py
import asyncio
from playwright.async_api import async_playwright
import json
import re
from urllib.parse import urljoin, urlparse
import os

class JobKoreaCSSCollector:
    def __init__(self):
        self.base_url = "https://www.jobkorea.co.kr"
        self.css_urls = set()
        self.inline_styles = []
        self.component_styles = {}
        self.color_palette = set()
        self.typography = {}
        
    async def extract_css_urls(self, page):
        """Extract all CSS file URLs from the page"""
        # Get <link> stylesheet URLs
        link_elements = await page.query_selector_all('link[rel="stylesheet"]')
        for link in link_elements:
            href = await link.get_attribute('href')
            if href:
                full_url = urljoin(self.base_url, href)
                self.css_urls.add(full_url)
        
        # Get @import URLs from style tags
        style_elements = await page.query_selector_all('style')
        for style in style_elements:
            content = await style.inner_text()
            import_matches = re.findall(r'@import\s+url\(["\']?([^"\']+)["\']?\)', content)
            for match in import_matches:
                full_url = urljoin(self.base_url, match)
                self.css_urls.add(full_url)
    
    async def extract_inline_styles(self, page):
        """Extract inline styles from style tags"""
        style_elements = await page.query_selector_all('style')
        for style in style_elements:
            content = await style.inner_text()
            if content.strip():
                self.inline_styles.append({
                    'url': page.url,
                    'content': content
                })
    
    async def extract_colors_from_styles(self, styles):
        """Extract color values from computed styles"""
        color_properties = ['color', 'background-color', 'border-color', 'border-top-color', 
                          'border-right-color', 'border-bottom-color', 'border-left-color']
        
        for prop in color_properties:
            if prop in styles and styles[prop] and styles[prop] != 'rgba(0, 0, 0, 0)':
                self.color_palette.add(styles[prop])
    
    async def extract_component_styles(self, page, component_name, selector):
        """Extract computed styles for a specific component"""
        try:
            elements = await page.query_selector_all(selector)
            if not elements:
                return
            
            # Get styles from first few instances
            component_styles = []
            for i, element in enumerate(elements[:3]):  # Limit to first 3 instances
                styles = await element.evaluate('''
                    (element) => {
                        const computed = window.getComputedStyle(element);
                        const styles = {};
                        
                        // Important style properties
                        const properties = [
                            'color', 'background-color', 'background-image',
                            'font-family', 'font-size', 'font-weight', 'line-height',
                            'padding', 'margin', 'border', 'border-radius',
                            'width', 'height', 'display', 'position',
                            'box-shadow', 'text-align', 'text-decoration'
                        ];
                        
                        properties.forEach(prop => {
                            styles[prop] = computed.getPropertyValue(prop);
                        });
                        
                        return styles;
                    }
                ''')
                
                await self.extract_colors_from_styles(styles)
                
                # Extract typography info
                if 'font-family' in styles:
                    font_key = f"{styles.get('font-family', '')}_{styles.get('font-size', '')}_{styles.get('font-weight', '')}"
                    self.typography[font_key] = {
                        'font-family': styles.get('font-family', ''),
                        'font-size': styles.get('font-size', ''),
                        'font-weight': styles.get('font-weight', ''),
                        'line-height': styles.get('line-height', '')
                    }
                
                component_styles.append(styles)
            
            self.component_styles[component_name] = component_styles
            
        except Exception as e:
            print(f"Error extracting styles for {component_name}: {e}")
    
    async def collect_page_styles(self, page, page_name):
        """Collect all styles from a specific page"""
        print(f"\nCollecting styles from {page_name}...")
        
        # Extract CSS URLs and inline styles
        await self.extract_css_urls(page)
        await self.extract_inline_styles(page)
        
        # Define components to analyze based on page
        if page_name == "main":
            components = {
                'header': 'header, .header, #header',
                'navigation': 'nav, .nav, .navigation, .gnb',
                'search_box': '.search, .search-box, .search-form, input[type="search"]',
                'main_banner': '.banner, .main-banner, .visual',
                'job_card': '.job-card, .recruit-item, .list-item, .job-item',
                'button_primary': '.btn, .button, .btn-primary, button',
                'footer': 'footer, .footer, #footer'
            }
        elif page_name == "job_list":
            components = {
                'filter_sidebar': '.filter, .sidebar, .search-filter',
                'job_listing': '.list-item, .job-item, .recruit-info',
                'pagination': '.pagination, .paging, .page-nav',
                'sort_options': '.sort, .order, .sorting'
            }
        elif page_name == "company":
            components = {
                'company_card': '.company-item, .corp-item, .company-info',
                'company_logo': '.logo, .company-logo, .corp-logo',
                'info_section': '.info, .company-detail, .corp-detail'
            }
        elif page_name == "job_detail":
            components = {
                'job_header': '.detail-header, .job-header, .recruit-header',
                'job_content': '.content, .detail-content, .job-detail',
                'apply_button': '.apply, .btn-apply, .apply-btn',
                'company_info': '.company-info, .corp-info'
            }
        else:
            components = {
                'general_card': '.card, .box, .panel',
                'form_input': 'input, textarea, select',
                'table': 'table, .table'
            }
        
        # Extract component styles
        for component_name, selector in components.items():
            await self.extract_component_styles(page, f"{page_name}_{component_name}", selector)
    
    async def run(self):
        """Main collection process"""
        async with async_playwright() as p:
            browser = await p.chromium.launch(headless=True)
            context = await browser.new_context(
                viewport={'width': 1920, 'height': 1080},
                user_agent='Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36'
            )
            page = await context.new_page()
            
            # Pages to visit
            pages_to_visit = [
                ('main', '/'),
                ('job_list', '/recruit/joblist'),
                ('company', '/company/companylist'),
                ('talent', '/talent/talentlist'),
                # We'll try to find a job detail page from the job list
            ]
            
            for page_name, path in pages_to_visit:
                try:
                    url = urljoin(self.base_url, path)
                    print(f"\nVisiting {page_name}: {url}")
                    await page.goto(url, wait_until='networkidle', timeout=30000)
                    await page.wait_for_timeout(2000)  # Wait for dynamic content
                    
                    await self.collect_page_styles(page, page_name)
                    
                    # If on job list page, try to get a job detail URL
                    if page_name == 'job_list':
                        try:
                            # Find first job link
                            job_link = await page.query_selector('a[href*="/recruit/jobinfo"]')
                            if job_link:
                                job_url = await job_link.get_attribute('href')
                                if job_url:
                                    full_job_url = urljoin(self.base_url, job_url)
                                    print(f"\nVisiting job detail: {full_job_url}")
                                    await page.goto(full_job_url, wait_until='networkidle', timeout=30000)
                                    await page.wait_for_timeout(2000)
                                    await self.collect_page_styles(page, 'job_detail')
                        except Exception as e:
                            print(f"Could not visit job detail page: {e}")
                    
                except Exception as e:
                    print(f"Error visiting {page_name}: {e}")
            
            await browser.close()
        
        # Save results
        self.save_results()
    
    def save_results(self):
        """Save collected data to files"""
        output_dir = r"D:\wspace\jobkorea_css\collected"
        
        # Save CSS URLs
        with open(os.path.join(output_dir, 'css_urls.txt'), 'w', encoding='utf-8') as f:
            f.write("# CSS File URLs from JobKorea\n\n")
            for url in sorted(self.css_urls):
                f.write(f"{url}\n")
        
        # Save inline styles
        with open(os.path.join(output_dir, 'inline_styles.json'), 'w', encoding='utf-8') as f:
            json.dump(self.inline_styles, f, indent=2, ensure_ascii=False)
        
        # Save component styles
        with open(os.path.join(output_dir, 'component_styles.json'), 'w', encoding='utf-8') as f:
            json.dump(self.component_styles, f, indent=2, ensure_ascii=False)
        
        # Save color palette
        color_list = list(self.color_palette)
        color_data = {
            'all_colors': sorted(color_list),
            'rgb_colors': [c for c in color_list if c.startswith('rgb')],
            'hex_colors': [c for c in color_list if c.startswith('#')],
            'named_colors': [c for c in color_list if not c.startswith(('#', 'rgb'))]
        }
        
        with open(os.path.join(output_dir, 'color_palette.json'), 'w', encoding='utf-8') as f:
            json.dump(color_data, f, indent=2, ensure_ascii=False)
        
        # Save typography
        with open(os.path.join(output_dir, 'typography.json'), 'w', encoding='utf-8') as f:
            json.dump(self.typography, f, indent=2, ensure_ascii=False)
        
        # Create summary report
        with open(os.path.join(output_dir, 'summary.txt'), 'w', encoding='utf-8') as f:
            f.write("JobKorea CSS Collection Summary\n")
            f.write("==============================\n\n")
            f.write(f"Total CSS URLs found: {len(self.css_urls)}\n")
            f.write(f"Inline style blocks: {len(self.inline_styles)}\n")
            f.write(f"Components analyzed: {len(self.component_styles)}\n")
            f.write(f"Unique colors found: {len(self.color_palette)}\n")
            f.write(f"Typography variations: {len(self.typography)}\n")
            
            f.write("\n\nComponents analyzed:\n")
            for component in sorted(self.component_styles.keys()):
                f.write(f"  - {component}\n")
        
        print("\n\nCollection complete! Results saved to:")
        print(f"  - {output_dir}/css_urls.txt")
        print(f"  - {output_dir}/inline_styles.json")
        print(f"  - {output_dir}/component_styles.json")
        print(f"  - {output_dir}/color_palette.json")
        print(f"  - {output_dir}/typography.json")
        print(f"  - {output_dir}/summary.txt")

if __name__ == "__main__":
    collector = JobKoreaCSSCollector()
    asyncio.run(collector.run())